website/public/js/controllers/inventoryCtrl.js

```diff
@@ -115,11 +115,12 @@ habitrpg.controller("InventoryCtrl",
       var potName = Content.hatchingPotions[potion.key].text();
       if (!$window.confirm(window.env.t('hatchAPot', {potion: potName, egg: eggName}))) return;
 
-      var userHasPet = user.items.pets[egg.key + '-' + potion.key] > 0;
+      var invalidPet = (user.items.pets[egg.key + '-' + potion.key] > 0) ||
```

Comment:
You should keep the userHasPet variable, turn this one into isPremiumPet. That makes it more legible.